mods/default/functions.lua

```diff
@@ -576,3 +576,118 @@ function default.can_interact_with_node(player, pos)
 
 	return false
 end
+
+
+--
+-- Stone/Ore API
+--
+
+default.registered_stones = {}
+default.registered_ores = {}
+
+local function check_stone_groups(groups_a, groups_b)
+	for fieldname, value in pairs(groups_a) do
+		if groups_b[fieldname] == value then
+			return true
+		end
+	end
+	return false
+end
+
+local function register_stone_with_ore(ore, stone)
+	-- set name, description, texture and groups
+	-- name format: 'modNameStone:stoneName_with_oreName'
+	-- for example: 'morestones:granite_with_iron'
+	local fullname = stone.name .. "_with_" .. ore.name
+
+	-- description format: 'Stonename Orename Ore' -> 'Granite Iron Ore'
+	ore.description = stone.description .. " " .. ore.description .. " Ore"
```

Comment:
Not required to recycle this table. Instead of a deep copy you could simply create a new definition table here.